app/main/views/drafts.py

```diff
@@ -139,6 +139,8 @@ def list_draft_services():
         framework = Framework.query.filter(
             Framework.slug == framework_slug
         ).first()
```

Comment:
It's a bit of a pain that you can't provide an override error message to `first_or_404`.